### PR TITLE
Make WebSocket actions non-blocking

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/WebSocketUtil.java [hunk 1]
```diff
@@ -96,12 +95,14 @@ public static void handleHandshake(WebSocketService wsService, WebSocketConnecti
             public void onSuccess(Session session) {
                 // TODO: Need to create new struct
                 BStruct webSocketEndpoint =
-                    BLangConnectorSPIUtil.createBStruct(
-                    wsService.getResources()[0].getResourceInfo().getServiceInfo().getPackageInfo().getProgramFile(),
-                    PROTOCOL_PACKAGE_HTTP, WebSocketConstants.WEBSOCKET_ENDPOINT);
+                        BLangConnectorSPIUtil.createBStruct(
```
Comment (irunika) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/WebSocketUtil.java]: Why we change this and add more lines of code? Is there a specific reason?

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/PushBinary.java [hunk 2]
```diff
@@ -44,20 +45,24 @@
                 @Argument(name = "data", type = TypeKind.BLOB)
         }
 )
-public class PushBinary extends BlockingNativeCallableUnit {
+public class PushBinary implements NativeCallableUnit {
```
[the remark this answers is not in the document]
Comment (riyafa) [reply stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/PushBinary.java]: Those methods of the `WebSocketAsyncRemoteEndpoint` don't return a `ChannelFuture`

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/websocketclientendpoint/Start.java [hunk 3]
```diff
@@ -81,7 +81,12 @@ public void execute(Context context) {
         HandshakeFuture handshakeFuture = clientConnector.connect(clientConnectorListener);
         handshakeFuture.setHandshakeListener(
                 new WsHandshakeListener(context, wsService, clientConnectorListener));
-        context.setReturnValues();
+        try {
+            handshakeFuture.sync();
+        } catch (InterruptedException e) {
+            throw new BallerinaConnectorException("Error occured: " + e.getMessage());
```
Comment (irunika) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/websocketclientendpoint/Start.java]: a typo **occured** should be **occurred**.

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/Close.java [hunk 4]
```diff
@@ -46,7 +46,9 @@
                 @Argument(name = "reason", type = TypeKind.STRING)
         }
 )
-public class Close extends BlockingNativeCallableUnit {
+public class Close
+        extends BlockingNativeCallableUnit { //This is blocking because of the limitations in the transport where
```
Comment (irunika) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/Close.java]: Can we inline this with class and add a comment on another line.
Also I think it is better to make this a TODO comment. So we can address this later

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/Ping.java [hunk 5]
```diff
@@ -36,13 +37,15 @@
...
         receiver = @Receiver(type = TypeKind.STRUCT, structType = WebSocketConstants.WEBSOCKET_CONNECTOR,
-                structPackage = "ballerina.http"),
+                             structPackage = "ballerina.http"),
         args = {
                 @Argument(name = "wsConnector", type = TypeKind.STRUCT),
                 @Argument(name = "data", type = TypeKind.BLOB)
         }
 )
-public class Ping extends BlockingNativeCallableUnit {
+public class Ping
```
Comment (irunika) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/Ping.java]: Same comment here. We can make extends inline and comment on another line.

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/PushBinary.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (irunika) [reply stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/websocketconnector/PushBinary.java]: Ack

**Pull request merged**